modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java

```diff
@@ -88,22 +97,37 @@
             }
 
             if (record) {
-                for (int i = 0; i < max.length; i++) {
-                    if (max[i] <= time) {
-                        System.arraycopy(max, i, max, i + 1, max.length - i - 1);
+                for (int i = 0; i < latencies.length; i++) {
+                    if (latencies[i] <= latency) {
+                        System.arraycopy(latencies, i, latencies, i + 1, latencies.length - i - 1);
+                        System.arraycopy(opStartTimes, i, opStartTimes, i + 1, opStartTimes.length - i - 1);
 
-                        max[i] = time;
+                        latencies[i] = latency;
+                        opStartTimes[i] = start;
 
                         break;
                     }
                 }
             }
 
             if (cnt % 1000 == 0)
-                log.info("APPLICATION_STREAMED " + cnt + " transactions [worst_latency=" + Arrays.toString(max) + "]");
+                log.info("APPLICATION_STREAMED " + cnt + " transactions [worst_latency=" + Arrays.toString(latencies) + "]");
+        }
+
+        List<String> result = new ArrayList<>();
+        Calendar calendar = Calendar.getInstance();
```

Comment:
Calendar? ORLY? Have you considered LocalDateTime?

Reply:
Fixed.